ios/UIViewController+RNScreens.m

```diff
@@ -0,0 +1,52 @@
+#import "UIViewController+RNScreens.h"
+#import "RNSScreenContainer.h"
+#import "RNSScreenStack.h"
+#import "RNSScreen.h"
+
+#import <objc/runtime.h>
+
+
+@implementation UIViewController (RNScreens)
+
+- (UIViewController *)reactNativeScreensChildViewControllerForStatusBarStyle
+{
+  UIViewController* lastViewController = [[self childViewControllers] lastObject];
+  if ([lastViewController isKindOfClass:[RNScreensNavigationController class]] || [lastViewController isKindOfClass:[RNScreensViewController class]] || [lastViewController isKindOfClass:[RNSScreen class]]) {
+    return lastViewController;
+  }
+  return [self reactNativeScreensChildViewControllerForStatusBarStyle];
+}
+
+- (UIViewController *)reactNativeScreensChildViewControllerForStatusBarHidden
+{
+  UIViewController* lastViewController = [[self childViewControllers] lastObject];
+  if ([lastViewController isKindOfClass:[RNScreensNavigationController class]] || [lastViewController isKindOfClass:[RNScreensViewController class]] || [lastViewController isKindOfClass:[RNSScreen class]]) {
+    return lastViewController;
+  }
+  return [self reactNativeScreensChildViewControllerForStatusBarHidden];
+}
+
+- (UIStatusBarAnimation)reactNativeScreensPreferredStatusBarUpdateAnimation
+{
+  UIViewController* lastViewController = [[self childViewControllers] lastObject];
+  if ([lastViewController isKindOfClass:[RNScreensNavigationController class]] || [lastViewController isKindOfClass:[RNScreensViewController class]] || [lastViewController isKindOfClass:[RNSScreen class]]) {
+    return lastViewController.preferredStatusBarUpdateAnimation;
+  }
+  return [self reactNativeScreensPreferredStatusBarUpdateAnimation];
```

Comment:
isn't this an infinite loop?

Reply:
It will call the base method, that is how the swizzling works (based on 
 https://nshipster.com/method-swizzling/#invoking-_cmd)